src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -458,5 +472,59 @@ private DeleteOrganisationResponse deleteUserProfile(Organisation organisation,
         return organisationRepository.findByStatus(status);
     }
 
+    @Override
+    public ResponseEntity<Object> addPaymentAccountsToOrganisation(PbaEditRequest pbaEditRequest,
+                                                      String organisationIdentifier, String userId) {
+        Optional<Organisation> organisation = Optional.ofNullable(
+                getOrganisationByOrgIdentifier(organisationIdentifier));
+
+        if (organisation.isEmpty()) {
+            log.error("{}:: {}", loggingComponentName, NO_ORG_FOUND_FOR_GIVEN_ID);
+            throw new ResourceNotFoundException(NO_ORG_FOUND_FOR_GIVEN_ID);
+        }
+
+        organisationIdentifierValidatorImpl.validateOrganisationIsActive(organisation.get());
+        professionalUserService.checkUserStatusIsActiveByUserId(userId);
+
+        Set<String> invalidPaymentAccounts = new HashSet<>();
+        Set<String> duplicatePaymentAccounts = new HashSet<>();
+        Set<String> validPaymentAccounts = new HashSet<>();
+
+        pbaEditRequest.getPaymentAccounts().forEach(pbaAccount -> {
```

Comment:
enhance this to pass list directly to validation methods rather than iteration.

Reply:
removed iteration and passing collection.